src/components/transactions/Faucet/CaptchaFaucetModalContent.tsx

```diff
@@ -0,0 +1,135 @@
+import { Box, Button, CircularProgress, Typography } from '@mui/material';
+import { useState } from 'react';
+import Turnstile from 'react-turnstile';
+import { ComputedReserveData } from 'src/hooks/app-data-provider/useAppDataProvider';
+import { useWeb3Context } from 'src/libs/hooks/useWeb3Context';
+import { selectCurrentReserves } from 'src/store/poolSelectors';
+import { useRootStore } from 'src/store/root';
+
+import { TxSuccessView } from '../FlowCommons/Success';
+import { DetailsNumberLine } from '../FlowCommons/TxModalDetails';
+
+export const CaptchaFaucetModalContent = ({ underlyingAsset }: { underlyingAsset: string }) => {
+  const { account, currentMarket, currentMarketData } = useRootStore();
+  const reserves = useRootStore((state) => selectCurrentReserves(state));
+
+  const { watchModeOnlyAddress } = useWeb3Context();
+
+  const [captchaToken, setCaptchaToken] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
+  const [captchaLoading, setCaptchaLoading] = useState<boolean>(true);
+  const [txHash, setTxHash] = useState<string>('');
+  const [error, setError] = useState<string>('');
+
+  const faucetUrl = `${process.env.NEXT_PUBLIC_API_BASEURL}/faucet`;
+  const siteKey = process.env.NEXT_PUBLIC_TURNSTILE_SITE_KEY as string;
+
+  const poolReserve = reserves.find(
+    (reserve) => reserve.underlyingAsset === underlyingAsset
+  ) as ComputedReserveData;
+
+  const defaultValue = valueToWei('1000', 18);
+  const mintAmount = mintAmountsPerToken[poolReserve.symbol.toUpperCase()]
+    ? mintAmountsPerToken[poolReserve.symbol.toUpperCase()]
+    : defaultValue;
+  const normalizedAmount = normalize(mintAmount, poolReserve.decimals);
+
+  const captchaVerify = (token: string) => {
+    setCaptchaToken(token);
+    setCaptchaLoading(false);
+  };
+
+  const faucet = async () => {
+    try {
+      setTxHash('');
+      setLoading(true);
+      setError('');
+      const response = await fetch(faucetUrl, {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          address: account,
+          captchaToken,
+          market: currentMarket,
+          tokenAddress: poolReserve.underlyingAsset,
+          tokenSymbol: poolReserve.symbol,
+          faucetAddress: currentMarketData.addresses.FAUCET,
+        }),
+      });
+      if (!response.ok) {
+        console.log(response);
+        throw new Error(response.statusText);
+      }
+      const data = await response.json();
+      setTxHash(data.msg);
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        setError(e.message);
+      } else {
+        setError('An error occurred trying to send the transaction');
+      }
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  if (txHash) {
+    return (
+      <TxSuccessView
+        txHash={txHash}
+        action={<Trans>Received</Trans>}
```

Comment:
Here i think or we should leave a hint to the user that it may take a few minutes to receive the tx. You will shortly receive {x} you can view the transaction here

Reply:
Not important can improve later